### PR TITLE
Console - application general migration - eighth part

### PR DESCRIPTION
**Pull request by jansobczynski**
## Issue

https://gravitee.atlassian.net/browse/APIM-XXX

## Description

Migrating application transfer ownership tab
## Additional context

Screenshot:
Screenshot 2023-12-21 at 12 56 43 PM

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://pr.team-apim.gravitee.dev/6281/console](https://pr.team-apim.gravitee.dev/6281/console)
      Portal: [https://pr.team-apim.gravitee.dev/6281/portal](https://pr.team-apim.gravitee.dev/6281/portal)
      Management-api: [https://pr.team-apim.gravitee.dev/6281/api/management](https://pr.team-apim.gravitee.dev/6281/api/management)
      Gateway v4: [https://pr.team-apim.gravitee.dev/6281](https://pr.team-apim.gravitee.dev/6281)
      Gateway v3: [https://pr.gateway-v3.team-apim.gravitee.dev/6281](https://pr.gateway-v3.team-apim.gravitee.dev/6281)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-cfuwnflopl.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/services-ngx/application-transfer-ownership.service.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { Inject, Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { Constants } from '../entities/Constants';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApplicationTransferOwnershipService {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/services-ngx/application-transfer-ownership.service.ts]: What do you think about rename for ApplicationMemberService or adding to it if already exists ?
(like for api with api-member)

File: gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.ts [hunk 2]
```diff
@@ -13,11 +13,123 @@
...
+            return EMPTY;
+          },
+        ),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => this.ngOnInit);
+  }
+
+  private initForm(defaultRole: any) {
+    this.form = new UntypedFormGroup({
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.ts:126]: Note required. But maybe we can start to Type ou forms 🤷‍♂️

File: gravitee-apim-console-webui/src/services-ngx/application-transfer-ownership.service.ts (hunk 1 above)
Comment (jansobczynski) [reply gravitee-apim-console-webui/src/services-ngx/application-transfer-ownership.service.ts]: Done

File: gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.ts (hunk 2 above)
Comment (jansobczynski) [reply gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.ts:126]: Done

**Review by ThibaudAV — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.html [hunk 3]
```diff
@@ -1,18 +1,76 @@
...
+          Application member
+        </mat-button-toggle>
+        <mat-button-toggle class="card__userOrGroup__toggle-group__toggle" value="user"> Other user </mat-button-toggle>
+        <mat-button-toggle class="card__userOrGroup__toggle-group__toggle" value="group"> Primary owner group </mat-button-toggle>
+      </mat-button-toggle-group>
+    </ng-container>
+
+    <div class="card__userOrGroup" *ngIf="form.get('userOrGroup').value === 'applicationMember'">
+      <mat-form-field class="card__userOrGroup__field">
+        <mat-label>Select Application member</mat-label>
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.html]:
```suggestion
        <mat-label>Select application member</mat-label>
```

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.html]: 🤷‍♂️

Comment (jansobczynski) [reply gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.html]: fixed

**Pull request merged**